### ISSUE TITLE
[Bug] Not all manifests present in bundle/manifests

### ISSUE THREAD

**Issue by grzesuav** [cryostatio/cryostat-operator#979]
### Current Behavior

I would like to install cryostat by applying manifests from https://github.com/cryostatio/cryostat-operator/tree/cryostat-v3.0/bundle/manifests

but not all manifests are present there, i.e. Deployment

### Expected Behavior

After applying manifests from https://github.com/cryostatio/cryostat-operator/tree/cryostat-v3.0/bundle/manifests via kubectl I have up and running cryostat-operator

### Environment

```Markdown
- OS: 
- Environment: 
- Version:
```

**Comment by andrewazores** [comment]
I think that's expected behaviour from what you're describing. The [bundle](https://quay.io/repository/cryostat/cryostat-operator-bundle) is meant to be installed from [OperatorHub](https://operatorhub.io/operator/cryostat-operator), and those manifests support the bundle installation. Once the Operator is installed this way it should have its own Deployment. Cryostat Deployments are dynamically created by the Operator, after you as the user create Cryostat CRs.

**Comment by andrewazores** [comment]
You might be able to turn the bundle manifests into a deployment manually (sidestepping the OperatorHub and OLM installation), but that would be a very unsupported configuration. If you want to go that route, the ClusterServiceVersion manifest does actually describe the Operator's own Deployment:

https://github.com/cryostatio/cryostat-operator/blob/67a03d814545f01cf3033210768c2e420bb83ecf/bundle/manifests/cryostat-operator.clusterserviceversion.yaml#L971

so you could try to extract that description and try to do that manually. But this is really not the way that Operators are intended to be installed and managed.

**Comment by grzesuav** [comment]
Ok, I was under impression this is static manifests set which can be used i.e. in argo or via standalone kubectl, so I was expecing to see https://github.com/cryostatio/cryostat-operator/blob/main/config/manager/manager.yaml

> so you could try to extract that description and try to do that manually. But this is really not the way that Operators are intended to be installed and managed.

Let's separate the topic how OLM manages operators - not everybody is on openshift using OLM in their cluster, I created an issue earlier - https://github.com/cryostatio/cryostat-operator/issues/901 and it seems there was no option proposed for standalop operator installation.

Regarding how operators should be installed, recently kubebuilder is introducing helm plugin to generate deployment bundle for operators https://github.com/kubernetes-sigs/kubebuilder/pull/4227 - and there is plenty of operators already shipping helm charts.

@andrewazores how do you see `cryostat-operator` installation in native kubernetes environment without OLM ?

**Comment by andrewazores** [comment]
> how do you see cryostat-operator installation in native kubernetes environment without OLM ?

The simple and unfortunate answer is: we have no support for this.

As I said in #901, we don't have a lot of resources (developer hours) to work on implementing yet another way to install Cryostat. Our primary targeted use case is OpenShift with OLM, but as much as possible I try to make sure that we remain generally *compatible* with other Kubernetes distributions or even non-Kubernetes environments. But, *compatibility* and *supported installation* are not the same. I would be very happy to help review anything like a community-developed "Helm chart for installing the Operator", or even "Operator static manifests", but these are not things that the business side prioritizes and so they are not things that I (or the other part-timers on the team) have the time to focus my efforts on.

**Comment by grzesuav** [comment]
got it, are you open to accept contribution in this area ?

 I believe I can make simple addition to generate static manifest bundle in first place

**Comment by andrewazores** [comment]
Absolutely, I would review and accept a contribution for such a thing :-)

Whatever it ends up being, please try to make it such that it can be automatically (re)generated rather than hand-written, or else it's very likely that it will end up falling out of date and out of sync. If it can be updated automatically then I can add it to the upstream release process at least and make sure that it gets updated at release points. Any testing would be on a best-effort basis.
